### ISSUE TITLE
Suggestion to use Poetry as dependency management and packaging tool

### ISSUE THREAD

**Issue by khelfen** [openego/eDisGo#234]
Hey :)

With some smaller projects I started to use [Poetry](https://python-poetry.org/) for packaging and dependency management. And I have to say it works really well and makes things easier.

[Here](https://towardsdatascience.com/lets-jump-on-the-poetry-bandwagon-d0b650de17b6) is an article about the benefits and [here](https://realpython.com/dependency-management-python-poetry/) is a good one about the basic usage.

To summarize the benefits:
* it combines dependency management and publishing into one tool and makes both things very easy
* it helps to avoid dependency conflicts
* with the `lock` file it assures that developers use the exactly same environment

As far as I can see it the switch to Poetry shouldn't bring much overhead.

What do you think about this @birgits and @AnyaHe ?

**Comment by khelfen** [comment]
As windows useres have to rely on conda for installation the usage of poetry is not an option at the moment
